--- binutils.orig/gas/as.c	2020-07-30 08:41:25.034236441 +0100
+++ binutils-2.35/gas/as.c	2020-07-30 08:41:52.197058041 +0100
@@ -103,7 +103,7 @@ int verbose = 0;
 int flag_dwarf_cie_version = -1;
 
 /* The maximum level of DWARF DEBUG information we should manufacture.  */
-unsigned int dwarf_level = 0;
+unsigned int dwarf_level = 3;
 
 #if defined OBJ_ELF || defined OBJ_MAYBE_ELF
 int flag_use_elf_stt_common = DEFAULT_GENERATE_ELF_STT_COMMON;
